src/algebra/scalar/authenticated_scalar.rs

```diff
@@ -1088,6 +1095,78 @@ impl<C: CurveGroup> Mul<&AuthenticatedScalarResult<C>> for &CurvePointResult<C>
 impl_borrow_variants!(CurvePointResult<C>, Mul, mul, *, AuthenticatedScalarResult<C>, Output=AuthenticatedPointResult<C>, C: CurveGroup);
 impl_commutative!(CurvePointResult<C>, Mul, mul, *, AuthenticatedScalarResult<C>, Output=AuthenticatedPointResult<C>, C: CurveGroup);
 
+// === FFT and IFFT === //
+impl<C: CurveGroup> AuthenticatedScalarResult<C>
+where
+    C::ScalarField: FftField,
+{
+    /// Compute the FFT of a vector of `AuthenticatedScalarResult`s
+    pub fn fft<D: EvaluationDomain<C::ScalarField>>(
+        x: &[AuthenticatedScalarResult<C>],
+    ) -> Vec<AuthenticatedScalarResult<C>> {
+        Self::fft_helper::<D>(x, true /* is_forward */)
+    }
+
+    /// Compute the inverse FFT of a vector of `AuthenticatedScalarResult`s
+    pub fn ifft<D: EvaluationDomain<C::ScalarField>>(
+        x: &[AuthenticatedScalarResult<C>],
+    ) -> Vec<AuthenticatedScalarResult<C>> {
+        Self::fft_helper::<D>(x, false /* is_forward */)
+    }
+
+    /// An FFT/IFFT helper that encapsulates the setup and restructuring of an FFT regardless of direction
+    ///
+    /// If `is_forward` is set, an FFT is performed. Otherwise, an IFFT is performed
+    fn fft_helper<D: EvaluationDomain<C::ScalarField>>(
+        x: &[AuthenticatedScalarResult<C>],
+        is_forward: bool,
+    ) -> Vec<AuthenticatedScalarResult<C>> {
+        assert!(!x.is_empty(), "Cannot compute FFT of empty vector");
+        let fabric = x[0].fabric();
+
+        // Extend to the next power of two
+        let n = x.len();
+        let padding_length = n.next_power_of_two() - n;
+        let pad = fabric.zeros_authenticated(padding_length);
+        let padded_input = [x, &pad].concat();
+
+        // Take the FFT of the shares and the macs separately
+        let shares = padded_input.iter().map(|v| v.share()).collect_vec();
+        let macs = padded_input.iter().map(|v| v.mac_share()).collect_vec();
+
+        let (share_fft, mac_fft) = if is_forward {
+            (
+                ScalarResult::fft::<D>(&shares),
+                ScalarResult::fft::<D>(&macs),
+            )
+        } else {
+            (
+                ScalarResult::ifft::<D>(&shares),
+                ScalarResult::ifft::<D>(&macs),
+            )
+        };
+
+        // No public values are added in an FFT, so the public modifiers remain unchanged
+        let modifiers = padded_input
+            .into_iter()
+            .map(|v| v.public_modifier)
+            .chain(iter::repeat(fabric.one()))
+            .take(n)
+            .collect_vec();
```

Comment:
Not a big deal, but couldn't we just do:
```suggestion
        let modifiers = x
            .into_iter()
            .map(|v| v.public_modifier)
            .collect_vec();
```

since `x` is of length n? or at least remove the `.chain` step?